tests/test_fluid_simulation.py

```diff
@@ -113,3 +108,315 @@ def test_fluid_simulation_phases(self, fluent_simulation):
+        #     phase Water at 25 C (2) Copper (3)
+        #  face_ids
+        #         1        9.9700e+02
+        #         2        9.9700e+02
+        #         3        9.9700e+02
+        #         4        9.9700e+02
+        #         5        9.9700e+02
+        #         6        9.9700e+02
+        #       ...               ...        ...
+        # """  # noqa: W291, E501
+        #         assert str(result) == ref
+        #         result.plot()
+        result = cfx_simulation.density_on_faces(face_ids=cfx_simulation.mesh.face_ids)
+        assert result.index.mesh_index.location == post.locations.faces
+        ref = """
+  results     RHO (kg*m^-3)           
+  set_ids                 1           
+    phase Water at 25 C (2) Copper (3)
+ face_ids                             
+"""  # noqa: W291, E501
+        assert str(result) == ref
+        with pytest.raises(
+            ValueError, match="Cannot plot a Dataframe with an empty mesh index."
+        ):
+            result.plot()
+
+    def test_results_cfx_cross_locations_on_cells(self, cfx_simulation):
+        result = cfx_simulation.density_on_cells(
+            cell_ids=cfx_simulation.mesh.element_ids
+        )
+        assert result.index.mesh_index.location == "cells"
+        ref = """
+  results     RHO (kg*m^-3)           
+  set_ids                 1           
+    phase Water at 25 C (2) Copper (3)
+ cell_ids                             
+        1        9.9700e+02           
+        2        9.9700e+02           
+        3        9.9700e+02           
+        4        9.9700e+02           
+        5        9.9700e+02           
+        6        9.9700e+02           
+      ...               ...        ...
+"""  # noqa: W291, E501
+        assert str(result) == ref
+
+    def test_results_fluent_cross_locations_on_nodes(self, fluent_simulation):
+        result = fluent_simulation.density_on_nodes(
+            node_ids=fluent_simulation.mesh.node_ids
+        )
+        assert result.index.mesh_index.location == post.locations.nodal
+        ref = """
+  results RHO (kg*m^-3)
+  set_ids             1
+ node_ids              
+        1    1.0742e+00
+        2    1.0436e+00
+        3    1.0131e+00
+        4    1.0327e+00
+        5    1.0247e+00
+        6    1.0445e+00
+      ...           ...
+"""  # noqa: W291, E501
+        assert str(result) == ref
+        result = fluent_simulation.density_on_nodes(
+            cell_ids=fluent_simulation.mesh.element_ids
+        )
+        assert result.index.mesh_index.location == post.locations.nodal
+        ref = """
+  results RHO (kg*m^-3)
+  set_ids             1
+ node_ids              
+      996    1.1041e+00
+      894    1.1035e+00
+      795    1.0982e+00
+      903    1.0985e+00
+      997    1.1097e+00
+      895    1.1091e+00
+      ...           ...
+"""  # noqa: W291, E501
+        assert str(result) == ref
+        result = fluent_simulation.density_on_nodes(
```

Comment:
@PProfizi Let's say that we have an `ElementalAndFaces` result and cell 30, face 25 and node 15 connected. Can you confirm the following workflows?

1. Ask for `density_on_cells` with cell 30 -> this would return the value on the file -> the result will be elemental -> **OK**
2. Ask for `density_on_cells` with node 15 -> this would do a transposition nodes->cells + get the value from the file -> the result will be elemental -> **OK**
3. Ask for `density_on_cells` with face 25 -> this would do a transposition faces->cells + get the value from the file -> the result will be elemental -> **OK**.
4. Ask for `density_on_nodes` with cell 30 -> this would do a get the value from the file +averaging to the nodes -> the result will be nodal -> **OK**.
5. Ask for `density_on_nodes` with node 15 ->This one is tricky, as the result is ElementalAndFaces. This would in my mind do a transposition nodes->cells + get the value from the file + averaging to nodes -> the result will be nodal -> **OK**.
6. Ask for `density_on_nodes` with face 25 -> This would get the value from the file + average to nodes -> **OK, However, this is the bug and we forbid it**.
8. Ask for `density_on_faces` with cell 30 -> This would do a transposition to faces + get the value from the file ->the result will be in faces -> **OK**.
9. Ask for `density_on_faces` with node 15 -> this would do a transposition nodes->faces + get the value from the file -> **OK, However, this is the bug and we forbid it**.
10. Ask for `density_on_faces` with face 25 -> Get the value from the file -> **OK, However, this is the bug and we forbid it**.

Reply:
@rafacanton 
1. Will add test
2. -> we do not currently allow querying cell result using node IDs
3. -> we do not currently allow querying cell result using face IDs
4. Will add test
5. Will check if this works
6. Forbidden as bugged
7. Will add test if not present
8. Not allowed for now
9. Forbidden as bugged

Basically I only allow querying results on "lower-dimension" entities with scopings, so cell_scoping for cell, face or nodal result, face_scoping for face or nodal result, node_scoping for nodal result

Reply:
@PProfizi Thanks for sharing the logic! Now I have the following question: Suppose that we fix the server bug and the bugged scenarios are allowed. Case 4 will yield a value at node 15 that would average the surrounding cell values. However, Case 6 will yield a value at node 15 that would average the surrounding face values. They will differ in general. This is expected, but should be communicated in the example that demonstrates the feature

Reply:
@rafacanton see [here](https://github.com/ansys/pydpf-post/blob/f76a819e99bc43c3ed27e7a62774d2ec32676185/src/ansys/dpf/post/fluid_simulation.py#L336)